windows/ReactNativeWebView/ReactWebView.cpp

```diff
@@ -55,94 +61,103 @@ namespace winrt::ReactNativeWebView::implementation {
                     self->OnNavigationFailed(sender, args);
                 }
             });
-
-        m_scriptNotifyRevoker = m_webView.ScriptNotify(
-            winrt::auto_revoke, [ref = get_weak()](auto const& sender, auto const& args) {
-                if (auto self = ref.get()) {
-                    self->OnScriptNotify(sender, args);
-                }
-            });
     }
 
-    void ReactWebView::WriteWebViewNavigationEventArg(winrt::IJSValueWriter const& eventDataWriter) {
-        auto tag = m_webView.GetValue(winrt::FrameworkElement::TagProperty()).as<winrt::IPropertyValue>().GetInt64();
-        WriteProperty(eventDataWriter, L"canGoBack", m_webView.CanGoBack());
-        WriteProperty(eventDataWriter, L"canGoForward", m_webView.CanGoForward());
-        WriteProperty(eventDataWriter, L"loading", !m_webView.IsLoaded());
+    void ReactWebView::WriteWebViewNavigationEventArg(winrt::WebView const& sender, winrt::IJSValueWriter const& eventDataWriter) {
+        auto tag = this->GetValue(winrt::FrameworkElement::TagProperty()).as<winrt::IPropertyValue>().GetInt64();
+        WriteProperty(eventDataWriter, L"canGoBack", sender.CanGoBack());
+        WriteProperty(eventDataWriter, L"canGoForward", sender.CanGoForward());
+        WriteProperty(eventDataWriter, L"loading", !sender.IsLoaded());
         WriteProperty(eventDataWriter, L"target", tag);
-        WriteProperty(eventDataWriter, L"title", m_webView.DocumentTitle());
-        if (auto uri = m_webView.Source()) {
-            WriteProperty(eventDataWriter, L"url", uri.AbsoluteCanonicalUri());
+        WriteProperty(eventDataWriter, L"title", sender.DocumentTitle());
+        if (auto uri = sender.Source()) {
+          WriteProperty(eventDataWriter, L"url", uri.AbsoluteCanonicalUri());
         }
     }
 
     void ReactWebView::OnNavigationStarting(winrt::WebView const& webView, winrt::WebViewNavigationStartingEventArgs const& /*args*/) {
         m_reactContext.DispatchEvent(
-            webView,
+            *this,
             L"topLoadingStart",
             [&](winrt::IJSValueWriter const& eventDataWriter) noexcept {
                 eventDataWriter.WriteObjectBegin();
-                WriteWebViewNavigationEventArg(eventDataWriter);
+                WriteWebViewNavigationEventArg(webView, eventDataWriter);
                 eventDataWriter.WriteObjectEnd();
             });
+
+        if (m_messagingEnabled) {
+          m_webBridge = WebBridge();
+          m_message_token = m_webBridge.MessagePostEvent([this](const auto&, hstring const& message)
+            {
+              this->OnMessagePosted(message);
+            });
+          webView.AddWebAllowedObject(L"__REACT_WEB_VIEW_BRIDGE", m_webBridge);
```

Comment:
this API is gone in WebView 2 which is soon going to be the standard (and the only option with WinUI 3). I would much rather have a solution that will be forward compatible and doesn't require webview to be re-written to be compatible.
@tero-paananen would it be possible to remove the dependency on this and use existing native module functionality instead of circumventing it this way?

Reply:
We will fix current code and use old WebView. After that i will try to start using WebView2 & WebUI 3